### ISSUE TITLE
[PoC] Cast from int256 to uint256 not safe

### ISSUE THREAD

**Issue by josojo** [gnosis/dex-contracts#293]
We need to make sure that this line is safe:
```
        uint256 burntFees = uint256(tokenConservation[0]) / 2;
```
and does not under/overflow

**Comment by nlordell** [comment]
If we check that the token conservation of the fee token is at least some amount in #292 then the casting should be safe. The division is always safe as long as it isn't by 0.

**Comment by bh2smith** [comment]
>  The division is always safe as long as it isn't by 0.

I agree that this does not need to be safe since it can never be unsafe.

@josojo How do you feel?

**Comment by josojo** [comment]
Yes, if we do the check proposed in  #292 , then it is safe. 
I noted it anyways, such that we do not forget, in case we decide for something else in  #292

**Comment by nlordell** [comment]
For now, we can check that tokenConservation for the fee token is positive in the token conservation check. This will make this casting safe and then we can modify it once we do something for #292?

**Comment by bh2smith** [comment]
Is this issue solved by #301?
